checksum/checksum.go

```diff
@@ -76,42 +93,22 @@ func (c *Checksum) SetTillichZemor(v [64]byte) {
 	checksum.SetSum(v[:])
 }
 
-// ToV2 converts Checksum to v2 Checksum message.
-//
-// Nil Checksum converts to nil.
-func (c *Checksum) ToV2() *refs.Checksum {
-	return (*refs.Checksum)(c)
-}
-
+// Equal return boolean value that means
+// the equality of the passed Checksums.
 func Equal(cs1, cs2 *Checksum) bool {
 	return cs1.Type() == cs2.Type() && bytes.Equal(cs1.Sum(), cs2.Sum())
 }
 
-// Marshal marshals Checksum into a protobuf binary form.
-func (c *Checksum) Marshal() ([]byte, error) {
-	return (*refs.Checksum)(c).StableMarshal(nil)
-}
-
-// Unmarshal unmarshals protobuf binary representation of Checksum.
-func (c *Checksum) Unmarshal(data []byte) error {
-	return (*refs.Checksum)(c).Unmarshal(data)
-}
-
-// MarshalJSON encodes Checksum to protobuf JSON format.
-func (c *Checksum) MarshalJSON() ([]byte, error) {
-	return (*refs.Checksum)(c).MarshalJSON()
-}
-
-// UnmarshalJSON decodes Checksum from protobuf JSON format.
-func (c *Checksum) UnmarshalJSON(data []byte) error {
-	return (*refs.Checksum)(c).UnmarshalJSON(data)
-}
-
-func (c *Checksum) String() string {
-	return hex.EncodeToString((*refs.Checksum)(c).GetSum())
+// String return Checksum string representation
```

Comment:
> formatless

does it mean not to document `String` methods at all? what format such method should return?

Reply:
I mean if type has fixed string format, it'd be better to implement `encoding.TextMarshaler/Unmarshaler` and make `String()` free and lightweight since it is used mostly in prints. About documenting I'd just write:
```
// String implements fmt.Stringer interface method.
```

Reply:
i just do not understand what "formatless" means, do you mean that it should not be documented and it should be ok to change the format?

and also I do not get do you prefer leaving it as `HexString` method or implementing `encoding.TextMarshaler`